crates/sui-protocol-config/src/lib.rs

```diff
@@ -32,6 +38,46 @@ impl ProtocolVersion {
     }
 }
 
+impl std::ops::Sub<u64> for ProtocolVersion {
+    type Output = Self;
+    fn sub(self, rhs: u64) -> Self::Output {
+        Self::new(self.0 - rhs)
+    }
+}
+
+impl std::ops::Add<u64> for ProtocolVersion {
+    type Output = Self;
+    fn add(self, rhs: u64) -> Self::Output {
+        Self::new(self.0 + rhs)
+    }
+}
+
+/// Models the set of protocol versions supported by a validator.
+/// The `sui-node` binary will always use the SYSTEM_DEFAULT constant, but for testing we need
+/// to be able to inject arbitrary versions into SuiNode.
+#[derive(Debug, Clone, Copy)]
+pub struct SupportedProtocolVersions {
+    min: ProtocolVersion,
+    max: ProtocolVersion,
```

Comment:
`std::ops::Range<ProtocolVersion>`?

Reply:
Unfortunately Range/RangeInclusive are not Copy, so I would have to litter all code with `.clone()` just to move a pair a `u64`s around.